ui/app/AppLayouts/Onboarding2/pages/KeycardAddKeyPairPage.qml

```diff
@@ -0,0 +1,219 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import StatusQ.Core 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+
+import AppLayouts.Onboarding2.controls 1.0
+import AppLayouts.Onboarding.enums 1.0
+
+OnboardingPage {
+    id: root
+
+    required property int addKeyPairState // Onboarding.AddKeyPairState.xxx
+    property int timeoutInterval: 30000
+
+    signal keypairAddContinueRequested()
+    signal keypairAddTryAgainRequested()
+    signal reloadKeycardRequested()
+    signal createProfilePageRequested()
+
+    pageClassName: "KeycardAddKeyPairPage"
+
+    Timer {
+        id: timer
+        interval: root.timeoutInterval
+        running: root.addKeyPairState === Onboarding.AddKeyPairState.InProgress
+        onTriggered: root.addKeyPairState = Onboarding.AddKeyPairState.Failed
+    }
+
+    states: [
+        State {
+            name: "inprogress"
+            when: root.addKeyPairState === Onboarding.AddKeyPairState.InProgress
+            PropertyChanges {
+                target: root
+                title: qsTr("Adding key pair to Keycard")
+            }
+            PropertyChanges {
+                target: iconLoader
+                sourceComponent: loadingIndicator
+            }
+            PropertyChanges {
+                target: subImageText
+                text: qsTr("Please keep the Keycard plugged in until the migration is complete")
+                visible: true
+            }
+            PropertyChanges {
+                target: image
+                source: Theme.png("onboarding/status_keycard_adding_keypair")
+            }
+        },
+        State {
+            name: "success"
+            when: root.addKeyPairState === Onboarding.AddKeyPairState.Success
+            PropertyChanges {
+                target: root
+                title: qsTr("Key pair added to Keycard")
+            }
+            PropertyChanges {
+                target: subtitle
+                text: qsTr("You will now require this Keycard to log into Status and transact with any accounts derived from this key pair")
+            }
+            PropertyChanges {
+                target: iconLoader
+                sourceComponent: successIcon
+            }
+            PropertyChanges {
+                target: continueButton
+                visible: true
+            }
+            PropertyChanges {
+                target: image
+                source: Theme.png("onboarding/status_keycard_adding_keypair_success")
+            }
+        },
+        State {
+            name: "failed"
+            when: root.addKeyPairState === Onboarding.AddKeyPairState.Failed
+            PropertyChanges {
+                target: root
+                title: "<font color='%1'>".arg(Theme.palette.dangerColor1) + qsTr("Failed to add key pair to Keycard") + "</font>"
```

Comment:
Wouldn't it be simpler to just change the color of the `Text` component instead?

Reply:
That's a `Page` `title` property, this is the only way